src/libs/Middleware/SaveResponseInOnyx.js

```diff
@@ -16,7 +35,7 @@ function SaveResponseInOnyx(response, request) {
 
         // For most requests we can immediately update Onyx. For write requests we queue the updates and apply them after the sequential queue has flushed to prevent a replay effect in
         // the UI. See https://github.com/Expensify/App/issues/12775 for more info.
-        const updateHandler = request.data.apiRequestType === CONST.API_REQUEST_TYPE.WRITE ? QueuedOnyxUpdates.queueOnyxUpdates : Onyx.update;
+        const updateHandler = request.data.apiRequestType === CONST.API_REQUEST_TYPE.WRITE ? QueuedOnyxUpdates.queueOnyxUpdates : updateOnyx;
```

Comment:
I think we should change both requests here, not only the usage of `Onyx.Update`.

I say that because write requests are the ones that actually generate an onyx response for all clients. If you call [OpenInitialSettings](https://github.com/Expensify/App/blob/d77564c05959933956e278f377fbca02c7906b2a/src/libs/actions/Wallet.js#L257), which is a read, we don't send data through onyx to it since it's just for the local client, it won't change anything. So that's not written in onyxUpdates.

But if you call [OpenReport](https://github.com/Expensify/App/blob/d77564c05959933956e278f377fbca02c7906b2a/src/libs/actions/Report.js#L520), which is a write and returns `lastUpdateID` and `previousUpdateID` (since we update the lastRead property of the report), we're not saving updating those properties locally

So let's say we're sending messages to someone, then we go from report A to report B and receive a new message, that would trigger GetOnyxUpdates, which is not ideal.

Reply:
Thanks for pointing that out. Makes sense!